File: CastIt.Client/lib/domain/models/dtos/responses/file_item_options_response_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'file_item_options_response_dto.freezed.dart';
part 'file_item_options_response_dto.g.dart';

@freezed
sealed class FileItemOptionsResponseDto with _$FileItemOptionsResponseDto {
  factory FileItemOptionsResponseDto({
    required int id,
    required bool isVideo,
    required bool isAudio,
    required bool isSubTitle,
    required bool isQuality,
    required String text,
    required bool isSelected,
    required bool isEnabled,
    String? path,
  }) = _FileItemOptionsResponseDto;

  factory FileItemOptionsResponseDto.fromJson(Map<String, dynamic> json) => _$FileItemOptionsResponseDtoFromJson(json);
}

File: CastIt.Client/ios/CastIt Watch App/Models/Responses/FileItemOptionsResponseDto.swift
import Foundation

// Mirrors ClientApp IFileItemOptionsResponseDto
struct FileItemOptionsResponseDto: Codable, Identifiable {
    let id: Int
    let isVideo: Bool
    let isAudio: Bool
    let isSubTitle: Bool
    let isQuality: Bool
    let path: String?
    let text: String
    let isSelected: Bool
    let isEnabled: Bool
}
